build.sbt

```diff
@@ -203,8 +161,119 @@ lazy val root = project
 
 lazy val noPublish = Seq(publishLocal := {}, publish := {}, publishArtifact := false)
 
+lazy val shared = Seq(
+  organization := "ch.epfl.bluebrain.nexus",
+  resolvers ++= Seq(
+    Resolver.bintrayRepo("bbp", "nexus-releases"),
+    Resolver.bintrayRepo("bbp", "nexus-snapshots")
+  )
+)
+
+lazy val compilation = Seq(
+  scalaVersion := "2.13.1", // scapegoat plugin not published yet for 2.13.2
+  // to be removed when migrating to 2.13.2 and replaced with @nowarn (scapegoat plugin not published yet for 2.13.2)
+  libraryDependencies ++= Seq(
+    compilerPlugin("com.github.ghik" % "silencer-plugin" % scalacSilencerVersion cross CrossVersion.full),
+    "com.github.ghik" % "silencer-lib" % scalacSilencerVersion % Provided cross CrossVersion.full
+  ),
+  scalacOptions ~= { options: Seq[String] => options.filterNot(Set("-Wself-implicit")) },
+  javacOptions ++= Seq(
+    "-source",
+    javaSpecificationVersion,
+    "-target",
+    javaSpecificationVersion,
+    "-Xlint"
+  ),
+  scalacOptions in (Compile, doc) ++= Seq("-no-link-warnings"),
+  javacOptions in (Compile, doc)  := Seq("-source", javaSpecificationVersion),
+  autoAPIMappings                 := true,
+  apiMappings += {
+    val scalaDocUrl = s"http://scala-lang.org/api/${scalaVersion.value}/"
+    ApiMappings.apiMappingFor((fullClasspath in Compile).value)("scala-library", scalaDocUrl)
+  },
+  // fail the build initialization if the JDK currently used is not ${javaSpecificationVersion} or higher
+  initialize := {
+    // runs the previous initialization
+    initialize.value
+    // runs the java compatibility check
+    val current  = VersionNumber(sys.props("java.specification.version"))
+    val required = VersionNumber(javaSpecificationVersion)
+    assert(CompatibleJavaVersion(current, required), s"Java '$required' or above required; current '$current'")
+  }
+)
+
+lazy val coverage = Seq(
+  coverageMinimum       := 80,
+  coverageFailOnMinimum := true
+)
+
+lazy val release = Seq(
+  bintrayOrganization := Some("bbp"),
+  bintrayRepository := {
+    import ch.epfl.scala.sbt.release.ReleaseEarly.Defaults
+    if (Defaults.isSnapshot.value) "nexus-snapshots"
+    else "nexus-releases"
+  },
+  sources in (Compile, doc)                := Seq.empty,
+  publishArtifact in packageDoc            := false,
+  publishArtifact in (Compile, packageSrc) := true,
+  publishArtifact in (Compile, packageDoc) := false,
+  publishArtifact in (Test, packageBin)    := false,
+  publishArtifact in (Test, packageDoc)    := false,
+  publishArtifact in (Test, packageSrc)    := false,
+  publishMavenStyle                        := true,
+  pomIncludeRepository                     := Function.const(false),
+  // removes compile time only dependencies from the resulting pom
+  pomPostProcess := { node =>
+    XmlTransformer.transformer(moduleFilter("org.scoverage") | moduleFilter("com.sksamuel.scapegoat")).transform(node).head
+  }
+)
+
+lazy val servicePackaging = {
```

Comment:
can't we move this somewhere out of the build.sbt?

It has way to many things.

Reply:
not really, no...